### ISSUE TITLE
Fix slider for smoother Up and Down cursor action

### ISSUE THREAD

**Issue by rbreaves** [Julow/Unexpected-Keyboard#364]
I added this to my qwerty layout and it appears to move the cursor way too fast up and down - while left and right are dialed in very well from the start it appears that adding up and down it does not really travel with the same level of control or a speed that feels natural.

```
<key width="3.9" key0="space" key7="switch_second" key5="left" key6="right" key7="up" key8="down" slider="true"/>
```

**Comment by Spike-from-NH** [comment]
@Julow states in #354 that slider is only implemented for left/right movement (key5 and key6). Like me, you are getting movement that isn't smooth because what you think is slider is merely auto-repeat. In closing that issue, I suggested slider be extended to the vertical. I feel the hard work has already been done!

Regarding "way too fast," you can control the repetition rate with the Unexpected config opfion, Key repeat interval.

**Comment by rbreaves** [comment]
Well initially I thought that but clearly I think slider just measures the relative speed given the distance traveled horizontally or vertically & given less vertical space you can’t have the same acceleration curve - they have to be different or you end up w/ what I was seeing - a wildly moving cursor up/down.

**Comment by Spike-from-NH** [comment]
Yes, that is how slider works—but your up-arrow/down-arrow key, again, is not slider!

I'd think a user would want an identical effect vertically, same number of characters generated for each mm the finger moves. Yes, the keyboard is wider than it is tall; I assume the program can retain focus even if a finger drags outside the program's screen territory. The space bar and its sliders are centrally located. Your vertical slider would need to be high enough to give the user room to slide southward.

**Comment by Julow** [comment]
I'll move this lower in my TODO list. Here's some pointers if someone is interested in implementing this:
- This piece of code decide on starting the slider mode and is specific for right and left movements: https://github.com/Julow/Unexpected-Keyboard/blob/master/srcs/juloo.keyboard2/Pointers.java#L260-L265
- This function handles move events when sliding and is also specific to one dimension: https://github.com/Julow/Unexpected-Keyboard/blob/master/srcs/juloo.keyboard2/Pointers.java#L415

**Comment by Julow** [comment]
I've done some improvements to the slider (7b8f739) and implemented up and down sliders (ca05c07, bffc769).
They can be added to a custom layout as `cursor_up` and `cursor_down` using the latest debug build: https://github.com/Julow/Unexpected-Keyboard/actions/runs/12735985454
I'd be happy to ear your feedback.

I've not added these to default layouts because of the undesirable behavior of Up and Down actions when moving past the ends of the text. Up and Down can move the cursor outside of the text box, focusing an other UI element and potentially closing the keyboard. I'd be happy to receive help on implementing something better.

**Comment by Spike-from-NH** [comment]
It should be easy to test to ensure that vertical sliding does not move the cursor beyond the bounds of the text being edited. If the slider simply emits Up Arrow or Down Arrow keystrokes, how could it leave the window? How does the horizontal slider avoid this undesirable behavior?

**Comment by Julow** [comment]
> It should be easy to test to ensure that vertical sliding does not move the cursor beyond the bounds of the text being edited.

It is not. I couldn't find any Android API that could easily tell the keyboard what is the vertical position of the cursor.
The slider simply emits Up and Down arrows, which can leave the text box. The horizontal slider works entirely differently, because a better API exists.

**Comment by Julow** [comment]
Have you been able to try out vertical sliders ? I was finding them too fast (like the original issue) so I made them accelerate less in https://github.com/Julow/Unexpected-Keyboard/pull/1023
I'd appreciate to ear what you think. Debug build at the bottom of this page: https://github.com/Julow/Unexpected-Keyboard/actions/runs/15956373831?pr=1023

**Comment by Julow** [comment]
With https://github.com/Julow/Unexpected-Keyboard/pull/1070 the up and down sliders seem quite usable. I'll close this issue. Please re-open if you have feedback or questions.
